### ISSUE TITLE
Node-RED is not updated if package lock exists

### ISSUE THREAD

**Issue by Steve-Mcl** [FlowFuse/device-agent#178]
### Current Behavior

When switching snapshots with different versions of Node-RED specified, "Installing Packages" is ran but Node-RED version remains old.
`npm install --production` (as the launcher and agent execute) ignore `package.json` and respect the lock file instead

### Expected Behavior

We should either we be inspecting the package or indiscriminately deleting the lock file in the agent device project dir when "Updating configuration files"-> writePackage() is called

### Steps To Reproduce

Only repeatable with new Deploy Snapshot to App Owned Devices.

### Environment

- FlowForge version:
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

**Comment by Pezmc** [comment]
`npm install --production` should always read the package.json and update the package-lock. So I think there must be something else at the root of why the NR package for example isn't updating.

It's `npm ci` that is intended to read the package lock and fail the command if the package-lock and package.json aren't in sync.

Testing locally:

```json5
// package.json
{
  "name": "testing",
  "dependencies": {
    "node-red": "^3.0.0"
  }
}
```

```shell
$ npm install
$ ls
node_modules/      package-lock.json  package.json
$ cat node_modules/node-red/package.json | jq .version
"3.0.0"
```


```json5
// Updated package.json
{
  "name": "testing",
  "dependencies": {
    "node-red": "^3.1.0"
  }
}
```

```shell
$ npm install --production
npm WARN config production Use `--omit=dev` instead.

up to date, audited 299 packages in 617ms

$ cat node_modules/node-red/package.json | jq .version
"3.1.0"
```

**Comment by Steve-Mcl** [comment]
@Pezmc thanks for taking the time to reason this out. 

I can only say, from what I witnessed what was occurring. And deleting package lock fixed the issue.

Perhaps it was an NPM version thing, or a windows thing, or a `~^>=` thing, or something else entirely.  The only thing I can say without re-testing and capturing more details is that deleting the lock, this stopped happening 100%.

---

_Brain engaged_

This might have been when changing stack from NR 3.1 to NR3.0.2.  But I was certain it was in both upgrade and downgrade situations.

**Comment by Pezmc** [comment]
If it was only for the _downgrade_, that could explain things, 3.1.0 is semantically compatible with 3.0.2, so it wouldn't downgrade the package.

That applies across the board, if you're putting `^3.0.2` in the package, you _could_ always end up with `3.1.0` or even `3.5.3` since they're both semantically compatible with `^3.0.2`. A downgrade is never guaranteed, even if you delete the package-lock file. 

We should perhaps swap to using specific version number or the tilde, to only pull in patches?

**Comment by MarianRaphael** [comment]
@Steve-Mcl is this fix in 1.14 or outstanding?

**Comment by Steve-Mcl** [comment]
Apologies, I thought I'd commented to let everyone know. Perhaps I did in slack and considered the matter parked?  

Anyhow, I was unable reliability make the issue reoccur. 

Consider this issue parked for now (feel free to close it)
